client/my-sites/site-settings/controller.js

```diff
@@ -118,31 +115,53 @@ module.exports = {
 	},
 
 	deleteSite( context ) {
-		let site = sites.getSelectedSite();
+		const { dispatch, getState } = context.store;
+		const siteSlug = getSelectedSiteSlug( getState() );
+		const siteId = getSelectedSiteId( getState() );
 
-		if ( sites.initialized ) {
-			if ( ! canDeleteSite( site ) ) {
-				return page( '/settings/general/' + site.slug );
+		const redirectIfCantDelete = () => {
```

Comment:
I guess that if I make this a controller property like you mention I'll end up with a new method that would be best if used in `index.js` as a middleware before `settingsController.deleteSite` and `settingsController.startOver`. Because the method still needs the context to get a new updated state by the time the listener passed with `ONCES_SITES_CHANGED` is called.